src/hooks/useKeyboardNavigation.js

```diff
@@ -0,0 +1,131 @@
+import PropTypes from 'prop-types';
+import { useEffect } from 'react';
+import tinykeys from 'tinykeys';
+
+const SEARCH_SECTION_INDEX = 1;
+
+const useKeyboardNavigation = ({
+  categoriesNavRef,
+  emojiSearchRef,
+  emojiListRef,
+}) => {
+  useEffect(() => {
+    return tinykeys(categoriesNavRef.current, {
+      ArrowLeft: focusPrevCategory,
+      ArrowRight: focusNextCategory,
+      ArrowDown: focusNextSection,
+    });
+  }, []);
+
+  useEffect(() => {
+    if (!emojiSearchRef.current) return;
+
+    insertSearchSection();
+
+    return tinykeys(emojiSearchRef.current, {
+      ArrowRight: focusSkinTonePicker,
+      ArrowUp: focusPrevSection,
+      ArrowDown: focusNextSection,
+    });
+  }, []);
+
+  useEffect(() => {
+    return tinykeys(emojiListRef.current, {
+      ArrowUp: focusPrevSection,
+      ArrowRight: focusNextEmoji,
+      ArrowLeft: focusPrevEmoji,
+    });
+  }, []);
+
+  const sections = [
+    {
+      name: 'categories',
+      focus: () => categoriesNavRef.current.firstChild.focus(),
+      rootElement: categoriesNavRef,
+    },
+    {
+      name: 'emoji list',
+      focus: () => {
+        const firstEmoji = emojiListRef.current.querySelector('.emoji');
+        firstEmoji.firstChild.focus();
+      },
+      rootElement: emojiListRef,
+    },
+  ];
+
+  const insertSearchSection = () => {
+    sections.splice(SEARCH_SECTION_INDEX, 0, {
+      name: 'search bar',
+      focus: () => emojiSearchRef.current.focus(),
+      rootElement: emojiSearchRef,
+    });
+  };
```

Comment:
Since this there is no direct mapping between the actual presentation and this list, having the `insertSearchSection` is a redundant function at the moment. Let's keep the list for now, though I do think that we can come up with a way to automagically create this list, but this can be done after we merge this.

Regardless, I think we can have it all in one place, and not have it in two places (both here and in the useEffect).

Last thing, I don't think we actually use the `name` anywhere. No need for it until we find a use for it. If you want it for readability, it can be a comment.

```suggestion
  const sections = [
    {
      // categories
      focus: () => categoriesNavRef.current.firstChild.focus(),
      rootElement: categoriesNavRef,
    },
    emojiSearchRef.current && {
      // search bar
      focus: () => emojiSearchRef.current.focus(),
      rootElement: emojiSearchRef,
    },
    {
      // emoji list
      focus: () => {
        const firstEmoji = emojiListRef.current.querySelector('.emoji');
        firstEmoji.firstChild.focus();
      },
      rootElement: emojiListRef,
    },
  ].filter(Boolean);
```

Reply:
Nice! I wasn't familiar with **.filter(Boolean);**